### ISSUE TITLE
Get List of Injectable Instance

### ISSUE THREAD

**Issue by tsctrl** [nestjs/nest#9514]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Hi, 

background:
I have issue to convert angular codes to nest js. Basically this part of codes are used to map two different model attribute. basic usage is to mapping response from other api response to nestjs model.

below is the part of the code:
```
export interface Converter<S, T> {
  convert(source: S, target?: T): T;
}

constructor(protected injector: Injector) { }
private converters: Map<InjectionToken<Converter<any, any>>, Converter<any, any>[]> = new Map();
private getConverters<S, T>(injectionToken: InjectionToken<Converter<S, T>>): Converter<S, T>[] {
    if (!this.converters.has(injectionToken)) {
      const converters = this.injector.get<Converter<S, T>[]>(injectionToken, []);
      if (!Array.isArray(converters)) {
        console.warn('Converter must be multi-provided, please use "multi: true" for', injectionToken.toString());
      }
      this.converters.set(injectionToken, converters);
    }
    return this.converters.get(injectionToken);
}

  /**
   * Pipeable operator to apply converter logic in a observable stream
   */
  pipeable<S, T>(injectionToken: InjectionToken<Converter<S, T>>): OperatorFunction<S, T> {
      return map((model: S) => this.convertSource(model, injectionToken));
  }

private convertSource<S, T>(source: S, injectionToken: InjectionToken<Converter<S, T>>): T {
    return this.getConverters(injectionToken).reduce((target, converter) => {
      return converter.convert(source, target);
    }, undefined as T);
}

// example converter implementation
@Injectable()
export class Normalizer implements Converter<MyModel, otherModel> {
  convert(source: MyModel, target: otherModel): otherModel {
    // do normalization/mapping code here
    return normalized;
  }

//create a token
export const NORMALIZER = new InjectionToken<Converter<any, MyModel>>('Normalizer');

//use the converter return new otherModel
this.apiService
      .get<MyModel>(url, { headers })
      .pipe(this.converter.pipeable(NORMALIZER));

```

To use the above codes i have to create a InjectionToken with the class required. and use pipe to convert the response from api response.

this work without any issue in angular. 

### Describe the solution you'd like

I am struggling to understand nestjs injector and module ref. getting list of injector token and how to implement this in nestjs. I have created injector token similar to angular with factory() method. and now i am puzzling how to replace the Injector in angular with similar solution in nestjs.

basically this part:
const converters = this.injector.get<Converter<S, T>[]>(injectionToken, []);

and the correct approach to create similar instance of injectionToken in angular in NestJs as below:
export const NORMALIZER = new InjectionToken<Converter<any, MyModel>>('Normalizer');

Please help. I am new to nestjs

Thanks at the earliest!

### Teachability, documentation, adoption, migration strategy

none

### What is the motivation / use case for changing the behavior?

This implementation provide good separation and best practice in angular and hoping i could port it to nestjs
